src/runtime/program/precompile_programs/secp256k1.zig

```diff
@@ -43,8 +49,595 @@ pub fn verify(
     current_instruction_data: []const u8,
     all_instruction_datas: []const []const u8,
 ) PrecompileProgramError!void {
-    _ = current_instruction_data;
-    _ = all_instruction_datas;
+    const data = current_instruction_data;
+
+    if (data.len < SECP256K1_DATA_START) {
+        if (data.len == 1 and data[0] == 0) return; // success
+        return error.InvalidInstructionDataSize;
+    }
+
+    const n_signatures = data[0];
+    if (n_signatures == 0 and data.len > 1) return error.InvalidInstructionDataSize;
+
+    const expected_data_size = SECP256K1_SIGNATURE_OFFSETS_START +
```

Comment:
these should be saturating? also shouldnt it be `n_sigs * serialized_size + 1`

Reply:
Used saturating - d62f9216225c38bf929ca75d4c6c26a2cecba13d
Also another bug - expected_data_size was a `u8`, oops